src/append/rolling_file/mod.rs

```diff
@@ -272,9 +288,44 @@ impl RollingFileAppenderBuilder {
             fs::create_dir_all(parent)?;
         }
 
-        // open the log file immediately
-        appender.get_writer(&mut appender.writer.lock())?;
+        // Create and open the log file immediately
+        if !&appender.path.exists() {
```

Comment:
I think the rest of this function would be expressed more clearly as an if/else chain with the body of the  `if let` extracted into an associated function that expresses the intent like `startup_roll`